src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -132,7 +132,32 @@ internal virtual string GetAssemblyFileVersion()
             {
                 var name = $"{Type.GetTypeInfo().Assembly.GetName().Name}.dll";
                 var filePath = Path.Combine(_clientDirectory, name);
-                return FileVersionInfo.GetVersionInfo(filePath).FileVersion;
+                var fileVersionInfo = FileVersionInfo.GetVersionInfo(filePath);
+
+                return $"{fileVersionInfo.FileVersion} ({ExtractShortSHA(fileVersionInfo.ProductVersion)})";
+            }
+
+            return "";
+        }
+
+        internal static string ExtractShortSHA(string productVersion)
+        {
+            if (productVersion != null)
+            {
+                string marker = "Commit Hash: "; // format used from AssemblyVersionAttribute
+                int found = productVersion.IndexOf(marker, StringComparison.CurrentCulture);
```

Comment:
@agocke Do we localize the "Commit Hash:" string? If not, this should probably be culture-invariant. (and if it is localized, then it's broken.)